### PR TITLE
Add VSCode config

### LINKED ISSUE #47: Add vscode config

### PR DESCRIPTION
**Pull request by AlexanderWells-diamond**
It's lifted from the DLS Py3 Template Module, with all auto-formatting parts removed due to most of the code not conforming to existing formatting (i.e. the Black formatter)

Fixes #47

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -20,6 +20,6 @@ softioc = {editable = true, extras = ["useful"], path = "."}
 [scripts]
 # Put coverage here so we don't interfere with debugging in the IDE
 # Put flake8 here so we don't run it during cibuildwheel tests
-tests = "python -m pytest --cov=softioc --cov-report term --flake8"
```
Comment (thomascobb) [inline Pipfile:23]: The CI for this is very different, so I put the flags here so we can separate flake8 from the pytest that is running in the container. If you change this you have to change the CI to match...

Comment (thomascobb) [reply Pipfile:23]: Cos we still need to run on python2 and I don't think flake8 works there...

Comment (thomascobb) [reply Pipfile:23]: or some other python2 reason that I can't remember

**Comment by AlexanderWells-diamond** [comment]
@thomascobb I see your latest commit _mostly_ fixed the problem. However, on Python3.9 I see these error messages:

```
Installing collected packages: pyparsing, tomli, toml, py, pluggy, packaging, iniconfig, coverage, attrs, typing-extensions, pytest, ply, nose, epicscorelibs, pytest-cov, pytest-asyncio, p4p, cothread, aioca
    Attempting uninstall: epicscorelibs
      Found existing installation: epicscorelibs 7.0.6.99.1.0
      Uninstalling epicscorelibs-7.0.6.99.1.0:
        Successfully uninstalled epicscorelibs-7.0.6.99.1.0
  Successfully installed aioca-1.2 attrs-21.2.0 cothread-2.17 coverage-6.0.1 epicscorelibs-7.0.6.99.2.0a1
```

The interesting part is: `epicscorelibs-7.0.6.99.2.0a1`. That's a [new alpha version](https://pypi.org/project/epicscorelibs/7.0.6.99.2.0a1/) that was only published  on Oct 7th. I do not understand how it is being picked up, as your `pyproject.toml` changes should pin to the previous stable version.

**Pull request merged**